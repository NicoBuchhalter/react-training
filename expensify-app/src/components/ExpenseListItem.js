import React from 'react';

export default ({ description, amount, createdAt}) => (
  <div>
    <h3>{description}</h3>
    <p>Amount: $ {parseFloat(amount / 100)}</p>
    <p>Created at: {createdAt}</p>
  </div>
); 