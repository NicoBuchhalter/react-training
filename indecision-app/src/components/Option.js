import React from 'react';

const Option = (props) => (
    <div className='option'>
        <p className='option__text'>{props.index}. {props.text}</p>
        <button 
            className='button--link'
            onClick={(e) => {
                props.handleDeleteOption(props.text);
            }}
        >
            Remove
        </button>
    </div>
);

export default Option;