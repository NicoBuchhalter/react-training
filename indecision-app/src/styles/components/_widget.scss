// Widget

.widget {
  background: $light-blue;
  margin-bottom: $xl-size;
}

.widget__message {
  color: $off-white;
  border-bottom: .1rem solid lighten($light-blue, 10%);
  padding: $l-size;
  margin: 0;
  text-align: center;
}



// Widget Header
.widget-header {
  background: $blue;
  color: $off-white;
  padding: $m-size;
  display: flex;
  justify-content: space-between;
}

.widget-header__title {
  margin: 0;
}