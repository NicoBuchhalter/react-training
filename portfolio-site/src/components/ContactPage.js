import React from 'react';

const ContactPage = () => (
  <div>
    <h3>Contact Me!</h3>
    <p>You can write an email to nico@test.com</p>
  </div>
);

export default ContactPage;