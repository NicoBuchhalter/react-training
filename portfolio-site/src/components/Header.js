import React from 'react';
import { NavLink } from 'react-router-dom';


const Header = () => (
  <header>
    <h1>Portfolio Site</h1>
    <NavLink  activeClassName="is-active" exact={true} to="/">Home Page</NavLink>
    <NavLink  activeClassName="is-active" exact={true} to="/portfolio">Portfolio</NavLink>
    <NavLink  activeClassName="is-active" exact={true} to="/contact">Contact</NavLink>
  </header>
);

export default Header;