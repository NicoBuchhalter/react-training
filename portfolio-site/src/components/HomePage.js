import React from 'react';

const HomePage = () => (
  <div>
    This is my dashboard component
  </div>
);

export default HomePage;