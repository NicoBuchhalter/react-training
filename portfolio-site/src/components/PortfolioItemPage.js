import React from 'react';


const PortfolioItemPage = (props) => {
  return (
    <div>
      This is portfolio number {props.match.params.id}
    </div>
  )
};

export default PortfolioItemPage;