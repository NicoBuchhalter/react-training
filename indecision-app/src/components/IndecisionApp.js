import React from 'react';
import Header from './Header';
import Options from './Options';
import Action from './Action';
import AddOption from './AddOption';
import OptionModal from './OptionModal';

export default class IndecisionApp extends React.Component {
    state = {
        options: [],
        selectedOption: undefined
    };

    handlePick = () => {
        const randomNum = Math.floor(Math.random() * this.state.options.length);
        this.setState(() => ({ selectedOption: this.state.options[randomNum] }));
    };
  
    handleDeleteOptions = () => {
        this.setState(() => ({ options: [] }));
    };
  
    handleDeleteOption = (option) => {
        this.setState((prevState) => ({
            options: prevState.options.filter((element) => element !== option)
        }));
    };
  
    handleAddOption = (option) => {
        if (!option) {
            return 'Enter valid value to add item.';
        } else if (this.state.options.indexOf(option) > -1) {
            return 'The item already exists.';
        }
        this.setState( (prevState) => ({ options: prevState.options.concat(option) }));
    };

    handleClearSelectedOption = () => {
        this.setState( () => ({ selectedOption: undefined }));
    };


  componentDidMount() {
      try {
          const options = JSON.parse(localStorage.getItem('options'));

          if (options) {
              this.setState(() => ({ options }));
          }
      }
      catch(e) {
          // Do nothing at all
      }
  }

  componentDidUpdate (prevProps, prevState){
      if (prevState.options.length !== this.state.options.length) {
          localStorage.setItem('options', JSON.stringify(this.state.options));
      }

  }

  render() {
      const subtitle = 'Put your life in the hands of a computer';
      return (
          <div>
              <Header subtitle={subtitle}/>
              <div className='container'>
                <Action 
                    hasOptions={this.state.options.length > 0}
                    handlePick={this.handlePick}
                />
                <div className='widget'>
                    <Options 
                        options={this.state.options}
                        handleDeleteOptions={this.handleDeleteOptions}
                        handleDeleteOption={this.handleDeleteOption}
                    />
                    <AddOption handleAddOption={this.handleAddOption}/>
                </div>
              </div>
              <OptionModal selectedOption={this.state.selectedOption} handleClearSelectedOption={this.handleClearSelectedOption}/>
          </div>
      );
  }
}

Header.defaultProps = {
  title: 'Indecision'
};