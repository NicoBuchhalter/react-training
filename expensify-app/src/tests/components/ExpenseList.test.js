import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseList } from '../../components/ExpenseList';
import expenses from '../fixtures/expenses';

test('Should render expense list with given expenses', () => {
  const wrapper = shallow(<ExpenseList expenses={expenses}/>);
  expect(wrapper).toMatchSnapshot();
});

test('Should render empty list message if no expenses', () => {
  const wrapper = shallow(<ExpenseList expenses={[]}/>);
  expect(wrapper).toMatchSnapshot(); 
});





