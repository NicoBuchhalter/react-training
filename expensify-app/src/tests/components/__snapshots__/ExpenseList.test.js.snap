// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render empty list message if no expenses 1`] = `
<div>
  <p>
    No Expenses
  </p>
</div>
`;

exports[`Should render expense list with given expenses 1`] = `
<div>
  <ExpenseListItem
    amount={750}
    createdAt={0}
    description="Gum"
    id="1"
    key="1"
    note=""
  />
  <ExpenseListItem
    amount={1090000}
    createdAt={-345600000}
    description="Rent"
    id="2"
    key="2"
    note=""
  />
  <ExpenseListItem
    amount={655000}
    createdAt={345600000}
    description="Credit Card"
    id="3"
    key="3"
    note=""
  />
</div>
`;
